### ISSUE TITLE
Support webworker

### ISSUE THREAD

**Issue by oeway** [gzuidhof/zarr.js#18]
I got this error: `ReferenceError: Buffer is not defined at new NestedArray (zarr.umd.js:2129)` when I try to use the [umd module](https://unpkg.com/zarr/dist/zarr.umd.js) in a webworker, could you please have a look?

**Comment by manzt** [comment]
This looks related to #5. My guess is that it has to do with a Buffer being specific to Node

**Comment by manzt** [comment]
https://github.com/gzuidhof/zarr.js/blob/f8bc95b7bef505b27e0ebceae3fe7099cb3b93ec/src/nestedArray/index.ts#L47-L57

**Comment by oeway** [comment]
Since there is no `window` defined in webworker, so it make sense that `typeof window === 'undefined' && Buffer.isBuffer(data)` trigger that error.

I think we need to add something like this:
```
const isWebWorker = typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope;

if ( typeof window === 'undefined' && !isWebWorker && Buffer.isBuffer(data))...
```
Ref: https://stackoverflow.com/a/18002694

**Comment by gzuidhof** [comment]
Yeah I think that's the problem, I didn't consider that webworkers won't
have access to window.

Really what we want to know here is whether we're in Node or not, I'm sure
there's a way to check that specifically

**Comment by gzuidhof** [comment]
This [SO answer](https://stackoverflow.com/a/4224668) is where I got the idea for the window check from. I think the next answer should fix it:

```javascript
function isNode() { return typeof "process" !== "undefined" && process.versions && process.versions.node; }
```

And a tiny performance improvement I guess would be computing it just once:
```javascript
const IS_NODE = typeof "process" !== "undefined" && process.versions && process.versions.node
```

**Comment by gzuidhof** [comment]
I pushed version 0.1.5 which should fix this issue (PR #20), I confirmed it works in the tests and in the browser. Could you close this issue when you confirm it works in a webworker?

**Comment by oeway** [comment]
I just tested it and it now works in a webworker, thanks!
